app/models/export.rb

```diff
@@ -45,35 +46,22 @@ def ready?
   end
 
   def self.find_or_create_export(format, groupe_instructeurs)
-    export = Export.find_for_format_and_groupe_instructeurs(format, groupe_instructeurs)
-
-    if export.nil?
-      export = Export.create(
-        format: format,
-        groupe_instructeurs: groupe_instructeurs
-      )
-    end
-
-    export
+    create_with(groupe_instructeurs: groupe_instructeurs)
+      .create_or_find_by(format: format, key: generate_cache_key(groupe_instructeurs))
   end
 
   def self.find_for_format_and_groupe_instructeurs(format, groupe_instructeurs)
-    export_including_gis = Export
-      .joins(:exports_groupe_instructeurs)
-      .where(
-        format: format,
-        exports_groupe_instructeurs: { groupe_instructeur: groupe_instructeurs }
-      )
-
-    export_including_gis.find do |export|
-      export.groupe_instructeurs.pluck(:id).sort == groupe_instructeurs.map(&:id).sort
-    end
+    find_by(format: format, key: generate_cache_key(groupe_instructeurs))
+  end
+
+  def self.generate_cache_key(groupe_instructeurs)
+    "#{groupe_instructeurs.first&.procedure_id}-#{groupe_instructeurs.map(&:id).sort.join('-')}"
```

Comment:
Le `first&` m'embête un peu : j'ai l'imrpession que si y'a aucun groupe instructeur (ce qui n'est jamais supposé arriver), on va générer une clef de cache qui sera `"nil"`, et que ça cassera très vite avec le 'unique'.

Je serais plutôt pour mettre un `raise` (ou laisser le code planter) si y'a aucun groupe instructeur.

```suggestion
    "#{groupe_instructeurs.first&.procedure_id}-#{groupe_instructeurs.map(&:id).sort.join('-')}"
```

Reply:
Il y a une validation sur la présence des groupes instructeurs. Donc le code va déjà planter s’il n'y a pas de groupes. Je suis ok pour enlever `&`